src/driver/amdxdna/amdxdna_drv.c

```diff
@@ -380,16 +400,39 @@ static int amdxdna_pmops_resume(struct device *dev)
 		}
 	}
 
-	XDNA_INFO(xdna, "hardware context resuming...");
+	XDNA_DBG(xdna, "hardware context resuming...");
 	list_for_each_entry(client, &xdna->client_list, node)
 		amdxdna_hwctx_resume(client);
 	mutex_unlock(&xdna->dev_lock);
 
 	return 0;
 }
 
+static int amdxdna_rpmops_suspend(struct device *dev)
+{
+	struct amdxdna_dev *xdna = pci_get_drvdata(to_pci_dev(dev));
+	int ret;
+
+	ret = amdxdna_pmops_suspend(dev);
```

Comment:
```suggestion
	mutex_lock(&xdna->dev_lock);
	WARN_ON(!list_empty(&xdna->client_list);
	ret = amdxdna_dev_suspend_nolock(xdna);
	mutex_unlock(&xdna->dev_lock);
```

The rumtime suspend and resume is called when all of the clients are closed. Those there is no need to do hwctx resume.

The amdxdna_dev_suspend_nolock() will call xdna->dev_info->ops->suspend() and this function can be reused by amdxdna_pmops_suspend().